src/logic/currencyValues/store/actions/fetchCurrencyValues.js

```diff
@@ -27,12 +25,11 @@ export const fetchCurrencyValues = (safeAddress: string) => async (dispatch: Red
     Object.entries(storedCurrencies).forEach((kv) => {
       const safeAddr = kv[0]
       const value = kv[1]
+
       const { currencyRate, selectedCurrency } = value
       batch(() => {
-        dispatch(setSelectedCurrency(safeAddr, selectedCurrency))
+        dispatch(setSelectedCurrency(safeAddr, selectedCurrency || AVAILABLE_CURRENCIES.USD))
         dispatch(setCurrencyRate(safeAddr, currencyRate))
-        dispatch(fetchCurrencyRate(safeAddr, selectedCurrency))
```

Comment:
I think that the `NaN` problem on the balances is related with this, why did you remove this line?

Reply:
because if you call an action fetchCurrencyValues with a safeAddress as an argument it shouldn't fetch the rates for all safes

Reply:
idk, i tried various scenarios like opening different safe first, then switching to a different one, creating/loading a safe works fine for me too. Could you please try to clear the storage and try again?

Reply:
I cleared the storage and now seems working fine, not sure what happened.